src/components/Profile/Profile.js

```diff
@@ -1,33 +1,52 @@
-import {useAuth0} from "@auth0/auth0-react";
-import React from "react";
+import React, {useEffect, useState} from "react";
 import "../Reused/reused.css";
 import UserInfo from "./UserInfo/UserInfo";
-import {Grid} from "@mui/material";
+import Grid from "@mui/material/Grid";
 import FavoriteAlbums from "./FavoriteAlbums/FavoriteAlbums";
 import Reviews from "../Reviews/Reviews";
-import users from "../TestData/users.json";
+import {Navigate} from "react-router-dom";
+import {useSelector} from "react-redux";
+import {getReviewsForUser} from "../../services/reviews-service";
 
 const Profile = () => {
-  const {user, isAuthenticated, isLoading} = useAuth0();
+  let {loggedInUser, loading} = useSelector(state => state.loggedInUserData);
+  const [reviewsLoading, setLoading] = useState(true);
+  const [reviewsData, updateReviewsData] = useState([]);
 
-  if (isLoading) {
+  const fetchReviewsData = async () => {
+    const reviews = await getReviewsForUser(loggedInUser.id);
+    updateReviewsData(reviews);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    if (!loading) {
+      setLoading(true);
+      updateReviewsData([]);
+      fetchReviewsData();
+      console.log(loggedInUser);
+    }
+  }, [loading, loggedInUser]);
+
+  if (loading) {
     return <div>Loading ...</div>;
-  }
-  const user_id = users[user.email].user_id;
-  return (
-      isAuthenticated && (
-          <Grid container spacing={2}>
-            <Grid item xs={3}>
-              <FavoriteAlbums user={user}/>
-            </Grid>
-            <Grid item xs={0.5}/>
-            <Grid item xs={7}>
-              <UserInfo user={user}/>
-              <Reviews id={user_id} idType="user"/>
-            </Grid>
+  } else if (loggedInUser === undefined) {
```

Comment:
I think? this should be fine. Just maybe need a second glance over it

Reply:
Good catch, loggedInUser shouldn't even be undefined (instead of not logged in it should be an empty object). I changed it to check for that and redirect in that case as well.

Reply:
This actually ended up being more complicated than I thought, I implemented a more comprehensive fix that should work.